src/js/smtpserver.c

```diff
@@ -1,19 +1,154 @@
 #include "smtpserver.h"
 
+jsval create_response(JSContext *cx, int code, const char* message, int disconnect) { 
+	jsval rcode;
+	jsval rmessage;
+	JSObject *obj;
+	jsval rdisconnect; 	// better name? 
+				
+	rcode = INT_TO_JSVAL(code);
+	rdisconnect = disconnect == TRUE ? JSVAL_TRUE : JSVAL_FALSE;
```

Comment:
I think `disconnect` can be tested directly and will properly evaluate to (C) false when it's set to `FALSE`.